bluemira/power_cycle/time.py

```diff
@@ -134,54 +97,119 @@ class PowerCycleScenario(PowerCycleTimeABC):
     def __init__(
         self,
         name,
-        pulse_set: Union[PowerCyclePulse, List[PowerCyclePulse]],
+        pulse_set: List[PowerCyclePulse],
         label=None,
     ):
-        self.pulse_set = self._validate_pulse_set(pulse_set)
-        durations_list = self._build_durations_list(self.pulse_set)
-        super().__init__(name, durations_list, label=label)
-
-    @staticmethod
-    def _validate_pulse_set(pulse_set):
-        """
-        Validate 'pulse_set' input to be a list of 'PowerCyclePulse'
-        instances.
-        """
-        pulse_set = validate_list(pulse_set)
-        for element in pulse_set:
-            PowerCyclePulse.validate_class(element)
-        return pulse_set
-
-    # ------------------------------------------------------------------
-    #  OPERATIONS
-    # ------------------------------------------------------------------
+        super().__init__(name, self._build_durations_list(pulse_set), label=label)
+        self.pulse_set = pulse_set
 
     def build_phase_library(self):
         """
         Returns a 'dict' with phase labels as keys and the phases
         themselves as values.
         """
-        pulse_set = self.pulse_set
-        phase_library = dict()
-        for pulse in pulse_set:
-            phase_library_for_pulse = pulse.build_phase_library()
-            phase_library = {**phase_library, **phase_library_for_pulse}
+        phase_library = {}
+        for pulse in self.pulse_set:
+            phase_library = {**phase_library, **pulse.build_phase_library()}
         return phase_library
 
     def build_pulse_library(self):
         """
         Returns a 'dict' with pulse labels as keys and the pulses
         themselves as values.
         """
-        pulse_library = {pulse.label: pulse for pulse in self.pulse_set}
-        return pulse_library
+        return {pulse.label: pulse for pulse in self.pulse_set}
+
+
+@dataclass
+class ScenarioConfig:
+    name: str
+    pulses: list
+    repetition: list
+
+
+@dataclass
+class PulseConfig:
+    name: str
+    phases: list
+    label: Optional[str] = None
+
+
+@dataclass
+class PhaseConfig:
+    name: str
+    logical: str
+    breakdown: list
+
+
+@dataclass
+class BreakdownConfig(BaseConfig):
+    """Breakdown Config"""
+
+
+@dataclass
+class Breakdown:
+    name: str
+    duration: ArrayLike
+
+
+class ScenarioConfigDescriptor:
+    """Scenario config descriptor for use with dataclasses"""
+
+    def __set_name__(self, _, name: str):
+        """Set the attribute name from a dataclass"""
+        self._name = "_" + name
+
+    def __get__(self, obj: Any, _) -> str:
+        """Get the scenario config"""
+        return getattr(obj, self._name)
+
+    def __set__(self, obj: Any, value: Union[dict, ScenarioConfig]):
+        """Set the scenario config"""
+        if not isinstance(value, ScenarioConfig):
+            value = ScenarioConfig(**value)
+
+        setattr(obj, self._name, value)
+
+
+class LibraryConfigDescriptor:
```

Comment:
A few reasons
 - Inheriting from python builtin types is nearly always a bad idea. You can easily run into weird edge cases because in the low levels of python type checking is not necessarily done in the same way.
 - We get typing more completely so it is much easier for an external person to learn through dataclasses.
 - The descriptor just gives you an interface the same as using the `property` decorator in a normal class. 
 - Its more concise (less code required), there are less mechanics that need to be implemented because dataclass does the work for us. As a side effect we need to test less and therefore maintain less.